projects/packages/my-jetpack/_inc/components/product-interstitial/index.jsx

```diff
@@ -58,20 +56,21 @@ export default function ProductInterstitial( {
 	const navigateToMyJetpackOverviewPage = useMyJetpackNavigate( '/' );
 
 	const clickHandler = useCallback(
-		checkout => {
-			const activateOrCheckout = () => ( bundle ? Promise.resolve() : activate() );
+		( checkout, selectedProductDetail ) => {
+			const activateOrCheckout = () =>
+				selectedProductDetail?.isBundle ? Promise.resolve() : activate();
 
 			activateOrCheckout().finally( () => {
-				const product = select( STORE_ID ).getProduct( slug );
-				if ( bundle ) {
+				const postActivationUrl = selectedProductDetail?.postActivationUrl;
+				const hasRequiredPlan = selectedProductDetail?.hasRequiredPlan;
+				const isFree = selectedProductDetail?.pricingForUi?.isFree;
+				const needsPurchase = ! isFree && ! hasRequiredPlan;
+
+				if ( selectedProductDetail.isBundle ) {
```

Comment:
```suggestion
				if ( selectedProductDetail?.isBundle ) {
```

Reply:
This should resolve any type errors when `selectedProtectDetail` does not have an `isBundle` property :+1:

Reply:
I think the force push re-introduced this.